src/Bundle/Grid/AbstractGrid.php

```diff
@@ -15,9 +15,32 @@
 
 use Sylius\Bundle\GridBundle\Builder\GridBuilder;
 use Sylius\Bundle\GridBundle\Builder\GridBuilderInterface;
+use Sylius\Component\Grid\Attribute\AsGrid;
 
 abstract class AbstractGrid implements GridInterface
 {
+    public static function getName(): string
+    {
+        if ($attribute = self::getAttributes()) {
+            return $attribute[0]->newInstance()->getName();
+        }
+
+        return static::class;
+    }
+
+    public function getResourceClass(): string
+    {
+        if ($attribute = self::getAttributes()) {
+            return $attribute[0]->newInstance()->getResourceClass();
+        }
+
+        throw new \LogicException(sprintf(
```

Comment:
we should add an exception namespace on Grid. But this need to be done in another PR.